static/app/views/monitors/components/checkInTimeline.tsx

```diff
@@ -38,35 +47,82 @@ function getBucketedCheckInsPosition(
 }
 
 export function CheckInTimeline(props: Props) {
-  const {bucketedData, start, end} = props;
+  const {bucketedData, start, end, timeWindow} = props;
+
+  function JobTickTooltip(jobTick: JobTickData) {
+    const {startTs, endTs, envMapping} = jobTick;
+    const {dateTimeProps} = timeWindowData[timeWindow];
+    const capturedEnvs = Object.keys(envMapping);
+    const representsSingleJob =
+      capturedEnvs.length === 1 &&
+      Object.values(envMapping[capturedEnvs[0]]).reduce(
+        (sum, count) => sum + count,
+        0
+      ) === 1;
+
+    return (
+      <Fragment>
+        <TooltipTimeLabel>
+          <DateTime date={startTs * 1000} {...dateTimeProps} />
+          {!representsSingleJob && (
+            <Fragment>
+              <Text>-</Text>
+              <DateTime date={endTs * 1000} {...dateTimeProps} />
+            </Fragment>
+          )}
+        </TooltipTimeLabel>
+        <StatusCountContainer>
+          {Object.entries(envMapping).map(([envName, statusCounts]) =>
+            Object.entries(statusCounts).map(
+              ([status, count]) =>
+                count > 0 && (
+                  <Fragment key={`${envName}:${status}`}>
+                    <StatusLabel status={status as CheckInStatus}>
```

Comment:
Is this happening because of how we typed the stats now actually? 😅

Reply:
im actually unsure :(( it was puzzling me a lot, but didn't want it to block this so just leaving it this for now, lemme add a TODO

Reply:
Yeah toss a TODO and I can look later